HW1/P6/P6.py

```diff
@@ -0,0 +1,219 @@
+import pyspark
+import re
+import random
+
+def word_match(line):
+    """Matches all words in a line.
+    To be used with flatMap, is it returns a tuple of all the words in a given line.
+    """
+    # Just one or more letters
+    a_word = re.compile(r"[\w']+")
+    # Match all the words in a given line
+    matches = a_word.findall(line)
+    
+    # findall returns a list, so...
+    return matches
+
+def filter_words(word):
+    """Function that uses regular expressions to filter out words that are made of
+    entirely capital letters, or entirely capital letters followed by a period, as per
+    the problem statement.
+    Note that the way we matched words, \w+, has already filtered out words made of only numbers.
+    """
+    
```

Comment:
I think some regex is necessary, since, for example:

```
In [1]: "TEST!".isupper()
Out[1]: True
```

This would imply the string should be excluded, but you actually want to keep it since it is not a word of entirely capital letters nor a word of entirely capital letters followed by a period.